libs/functions/generate_api.py

```diff
@@ -156,9 +160,19 @@ def generate_region_timeseries(
             for row in model_output.data.to_dict(orient="records")
         ]
 
+    metrics_timeseries = []
+    region_level_metrics = calculate_top_level_metrics_for_timeseries(
```

Comment:
Can you pass this in as an argument (moving the calculation to `api_pipeline.py` i believe? I'd like to keep most of the burden of calculations out of the `generate_api` code.

Reply:
That seems a little weird to me given that here is where we also calculate `actuals_timeseries` which gets returned in `RegionSummaryWithTimeseries`. Makes sense to me have those two values evaluated in the same place.

Reply:
Ah, the actuals calculations does do that one calculation, but that is more of a relic that we will hopefully remove (that field has been deprecated for a while now).  

My thinking is that i'd generally like this file to be simple mappings to help make testing easier/separate how the data is persisted from how we're generating the data.

For example, I could see wanting to perform the metrics calculations in bulk to speed it up.  If we pass in just the metrics df for one fips here as an argument, I think it gives us more flexibility to do that sort of change outside of this code.